### DISCUSSION TITLE
Update many with different values

### DISCUSSION THREAD

**Discussion by andreasvh-conceto** [prisma/prisma#20000] [Q&A] [answered]
Hello,
is there a possibility to perform the following update query without a query raw? It has to be a bulk operation in one go for multiple items.  In postgres sql it is something like:

```sql
UPDATE your_table
SET column_name = new_values.new_value
FROM (
    VALUES 
        (id_1, value_1),
        (id_2, value_2),
        (id_3, value_3)
) AS new_values (id, new_value)
WHERE your_table.id = new_values.id;

```
Or do I have a chance to write a client extension for an updateMany method, which is able to perform such a query in a handy way?

Thanks for your help 👍 
Best regards
Andreas

**Comment by nurul3101** [comment, marked as answer]
Hello @andreasvh-conceto 👋 

You can use [interactive transactions](https://www.prisma.io/docs/guides/performance-and-optimization/prisma-client-transactions-guide#interactive-transactions) to perform multiple update operations in one go, but it is a bit different from the SQL query you provided, which performs a bulk update in a single query.

Though the `$transaction` API will still execute multiple queries within a single transaction.

It could look something like this:
```ts
await prisma.$transaction(async (prisma) => {
    await prisma.yourTable.update({ where: { id: id_1 }, data: { column_name: value_1 } });
    await prisma.yourTable.update({ where: { id: id_2 }, data: { column_name: value_2 } });
});
```

Hope this helps!
